tests/untl_structure_test.py

```diff
@@ -0,0 +1,601 @@
+    # Create an element with collection and meta fields.
+    root = us.UNTLElement()
+    root.tag = 'metadata'
+    root.contained_children = ['collection', 'meta']
+    collection = us.UNTLElement(content=u'Colección')
+    collection.tag = 'collection'
+    root.add_child(collection)
+    meta = us.UNTLElement(content='fake',
+                          qualifier='ark')
+    meta.tag = 'meta'
+    root.add_child(meta)
+    # NOTE: in utf-8 length is 94:
+    # "{'meta': [{'qualifier': 'ark', 'content': 'fake'}], 'collection': [{'content': 'Colección'}]}"  # noqa 
+    # We are getting that it is length 97 for:
+    # "{'meta': [{'content': 'fake', 'qualifier': 'ark'}], 'collection': [{'content': u'Colecci\xf3n'}]}"  # noqa
+    assert root.record_length == 97
+
+
+def test_UNTLElement_record_content_length():
+    """Check the record_length with meta field excluded."""
+    # Create an element with collection and meta fields.
+    root = us.UNTLElement()
+    root.tag = 'metadata'
+    root.contained_children = ['collection', 'meta']
+    collection = us.UNTLElement(content=u'Colección')
+    collection.tag = 'collection'
+    root.add_child(collection)
+    meta = us.UNTLElement(content='fake',
+                          qualifier='ark')
+    meta.tag = 'meta'
+    root.add_child(meta)
+    # NOTE: in utf-8 length is 42:
+    # "{'collection': [{'content': 'Colección'}]}"
+    # We are getting that it is length 46 for:
+    # "{'collection': [{'content': u'Colecci\xf3n'}]}"
+    assert root.record_content_length == 46
+
+
+@patch('pyuntl.untl_structure.FormGenerator.get_vocabularies', return_value=VOCAB)
+def test_FormGenerator(mock_vocab):
+    """Test same UNTLElement subclasses are grouped together.
+
+    Official and series titles should be in same group. A hidden
+    group should be created if it doesn't exist.
+    """
+    creator = us.Creator(qualifier='')
+    official_title = us.Title(content='A Title', qualifier='officialtitle')
+    series_title = us.Title(content='Series', qualifier='seriestitle')
+    children = [official_title, series_title, creator]
+    form_elements = us.FormGenerator(children=children,
+                                     sort_order=['title', 'creator', 'hidden'])
+    # Check there is a title group with both official and series elements
+    # a group for the creator, and a separate hidden group was created.
+    assert len(form_elements.element_groups) == 3
+    title_group = form_elements.element_groups[0].group_list
+    assert len(title_group) == 2
+    for title in title_group:
+        assert isinstance(title, us.Title)
+    creator_group = form_elements.element_groups[1].group_list
+    assert len(creator_group) == 1
+    assert isinstance(creator_group[0], us.Creator)
+    hidden_group = form_elements.element_groups[2].group_list
+    assert len(hidden_group) == 1
+    assert isinstance(hidden_group[0], us.Meta)
+    mock_vocab.assert_called_once_with()
```

Comment:
I don't know that checking if the mocked vocab getter has been called really adds anything to these tests. I _do_ want the mock, I just don't think it really needs to be checked.